## Analise dos dados coletados do site da Honda

## Importando as bibliotecas que auxiliaram na analise

In [14]:
import pandas as pd #principal biblioteca de análise de dados
import plotly.express as px #biblioteca responsavel por criar os graficos para a análise

In [3]:
#lendo a base de dados
df=pd.read_csv("Motos_honda.csv")
df

Nome nacionalidade  categoria  \
0                             CG 160 Start      Nacional     Street   
1                               CG 160 Fan      Nacional     Street   
2                             CG 160 Titan      Nacional     Street   
3                             CG 160 Cargo      Nacional     Street   
4                              Pop 110i ES      Nacional     Street   
5                                  Biz 125      Nacional     Street   
6                                Elite 125      Nacional     Street   
7                                      PCX      Nacional     Street   
8                                Honda ADV      Nacional     Street   
9                                    X-ADV      Nacional     Street   
10                         CB 300F Twister      Nacional     Street   
11                                 CB 500F      Nacional     Street   
12                                 CB 650R      Nacional     Street   
13                                CB 1000R      Nacional     Street   
14                  CB 1000R Black Edition      Nacional     Street   
15                         XR 300L Tornado      Nacional  Adventure   
16                            NXR 160 Bros      Nacional  Adventure   
17                              Sahara 300      Nacional  Adventure   
18                                 XRE 190      Nacional  Adventure   
19                                 CB 500X      Nacional  Adventure   
20                                 NC 750X     Importado  Adventure   
21                   CRF 1100L Africa Twin      Nacional  Adventure   
22  CRF 1100L Africa Twin Adventure Sports      Nacional  Adventure   
23                               Honda ADV      Nacional     Street   
24                                   X-ADV      Nacional     Street   
25                                CRF 250F      Nacional   Off Road   
26                           Linha CRF 250     Importado   Off Road   
27                           Linha CRF 450     Importado   Off Road   
28                        TRX 420 FourTrax      Nacional   Off Road   
29                                CBR 650R      Nacional      Sport   
30               CBR 1000RR-R FIREBLADE SP     Importado      Sport   
31                  GL 1800 Gold Wing Tour     Importado    Touring   

                                          cor cilindradas      preços  
0      Prata Metálico - Force Silver Metallic    162,7 cc   R$ 14.650  
1       Cinza Metálico - Embare Gray Metallic    162,7 cc   R$ 16.070  
2        Azul Perolizado - Pearl Spencer Blue    162,7 cc   R$ 17.440  
3                         Branco - Ross White    162,7 cc   R$ 15.970  
4                         Branco - Ross White    109,5 cc    R$ 9.690  
5                       Vermelho - Maceio Red   123,9 cc.   R$ 12.000  
6       Azul Metálico - Mystery Blue Metallic    123,9 cc   R$ 12.966  
7          Cinza Metálico - Una Gray Metallic    156,9 cc   R$ 16.800  
8     Verde Fosco - Mat Hunter Green Metallic    149,3 cc   R$ 23.060  
9      Cinza Metálico - Iridium Grey Metallic      745 cc   R$ 93.500  
10       Cinza Metálico - Cynos Gray Metallic    293,5 cc   R$ 22.370  
11       Cinza Fosco – Mat Crow Gray Metallic      471 cc   R$ 40.660  
12  Laranja Perolizado - Pearl Damasco Orange      649 cc   R$ 52.590  
13  Vermelho Metálico – Bordeaux Red Metallic    998,4 cc   R$ 78.870  
14       Azul Fosco – Mat Jeans Blue Metallic    998,4 cc   R$ 87.730  
15                    Vermelho - Fighting Red    293,5 cc   R$ 27.690  
16                    Vermelho - Fighting Red    162,7 cc   R$ 20.490  
17                             Cinza Metálico    293,5 cc   R$ 28.170  
18  Azul Metálico - Mat Noturno Blue Metallic    184,4 cc   R$ 22.006  
19                   Verde – Mat Hunter Green       471cc   R$ 43.260  
20                     Vermelho - Victory Red      745 cc   R$ 52.680  
21   Azul Metálico - Glint Wave Blue Metallic    1.084 cc   R$ 81.100  
22      Branco Perolizado - Pearl Glar

## Tratamento dos dados da base de dados

In [8]:
#Os dados do preços precisam ser transformado em float para podermos fazer as operações e analises

df["preços"]=df["preços"].str.replace("R$ ", "") #retirar os R$ dos preços
df["preços"]=df["preços"].str.replace(".", "") #Retirar o ponto nos números
df["preços"]=df["preços"].astype(float) #transformando o número que está em string para float

## Analise dos dados

### Nacionalidade das motos da honda

In [15]:
nacionalidade_preferencia=df["nacionalidade"].value_counts().reset_index() # Analise que irá contar a quantidade de cada nacionalidade que existe na base de dados
nacionalidade_preferencia_fig=px.pie(nacionalidade_preferencia, names="nacionalidade", values="count", hole=0.3)# Criando o grafico em pizza para uma melhor visualização
nacionalidade_preferencia_fig

### Categoria mais cara das motos da honda

In [23]:
#Analisando a média de preço de cada categoria
categoria_maior_media=round(df[["categoria", "preços"]].groupby("categoria").mean().sort_values(by="preços", ascending=False).reset_index(), 2) 

#Grafico em barra para uma melhor visualização
categoria_maior_media_fig=px.bar(categoria_maior_media, x="categoria", y="preços")
categoria_maior_media_fig

In [24]:
df["cilindradas"].value_counts() #Analise feita para descobrir qual das cilondradas é mais frequente nas moto da Honda

cilindradas
162,7 cc     5
745 cc       3
293,5 cc     3
1.084 cc     2
649 cc       2
998,4 cc     2
149,3 cc     2
156,9 cc     1
471 cc       1
123,9 cc     1
109,5 cc     1
184,4 cc     1
471cc        1
123,9 cc.    1
249,58 cc    1
249,4cc      1
449,7 cc     1
420 cc       1
1.000 cc     1
1.833 cc     1
Name: count, dtype: int64

In [29]:
media_precos=px.box(df["preços"]) #Grafico boxplot que  irá mostrar a distribuição dos preços de motos na Honda, a media, o míninmo, o maximo de preços e os outliers
media_precos





### **Relatório de Análise das Motos Honda**

#### **1. Distribuição das Motos da Honda por Origem (Gráfico de Pizza)**
As motos brasileiras são as mais frequentes na frota de motos da Honda. Isso pode ser visualizado no gráfico de pizza, que demonstra a distribuição percentual de motos por nacional ou importado.


#### **2. Categoria Mais Cara: Touring (Gráfico de Barras)**
A categoria mais cara entre as motos da Honda são as Touring, também conhecidas como "Tornados". O gráfico de barras compara os preços médios das motos por categoria, destacando a categoria Touring como a mais dispendiosa.


#### **3. Cilindradas Mais Comuns nas Motos Honda**
As cilindradas mais comuns entre as motos da Honda são:
- 162,7 cc
- 745 cc
- 293,5 cc

Essas cilindradas dominam o mercado, abrangendo a maioria dos modelos vendidos.

#### **4. Preço Médio, Mínimo e Máximo**
- **Preço médio das motos**: R$ 34.415,00
- **Preço mais baixo**: R$ 9.690,00
- **Preço mais alto**: R$ 304.450,00

Esses dados refletem a variação de preços entre as motos da Honda, demonstrando a ampla gama de opções oferecidas pela marca, desde modelos mais acessíveis até motos premium.

---

### **Conclusão**
Os dados apresentados confirmam que as motos brasileiras são predominantes na frota da Honda, e que a categoria Touring é a mais cara. Além disso, as cilindradas mais comuns indicam uma diversidade de modelos que atendem diferentes segmentos de mercado, com uma grande variação de preços, desde opções econômicas até motocicletas de alto desempenho.

